lib/standards/aria-roles.js

```diff
@@ -0,0 +1,221 @@
+// Source: https://www.w3.org/TR/wai-aria-1.1/#roles
+
+/* easiest way to see allowed roles is to filter out the global ones
+   from the list of inherited states and properties. The dpub spec
+   does not have the global list so you'll need to copy over from
+   the wai-aria one:
+
+  const globalAttrs = Array.from(
+    document.querySelectorAll('#global_states li')
+  ).map(li => li.textContent.replace(/\s*\(.*\)/, ''));
+
+  const globalRoleAttrs = Array.from(
+    document.querySelectorAll('.role-inherited li')
+  ).filter(li => globalAttrs.includes(li.textContent.replace(/\s*\(.*\)/, '')))
+
+  globalRoleAttrs.forEach(li => li.style.display = 'none');
+*/
+const ariaRoles = {
+	alert: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded'],
+		nameFrom: ['author']
+	},
+	alertdialog: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded', 'aria-modal'],
+		nameFrom: ['author']
+	},
+	application: {
+		type: 'landmark',
+		allowedAttrs: ['aria-activedescendant'],
+		nameFrom: ['author']
+	},
+	article: {
+		type: 'structure',
+		allowedAttrs: ['aria-posinset', 'aria-setsize', 'aria-expanded'],
+		nameFrom: ['author'],
+		implicit: ['article']
```

Comment:
We shouldn't need this anymore now that we have `implicitHtmlRole`. Looking through the code base, I think we need to update `aria.implicitNodes()`. It's a little hard to tell, but from the look of it that's the only place we're directly using this prop.